Minor DB Changes

RetroSabian opened the pull request:
Added MembershipsController and edited Membership for front end compatibility.

RetroSabian committed:
Added MembershipsController and edited Membership for front end compatibility.

DreyerChris commented on Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Controllers/MembershipsController.cs:
```diff
@@ -0,0 +1,36 @@
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Memberships>>> Get()
```
Its better to give classes singular names, `Memberships` -> `Membership`, otherwise the table name created by Entity Framework might be weird

RetroSabian replied:
I was advised to use this way, to change would be quiet taxing, is the naming convention that poor that it has to be changed?

grootendorste replied:
I see that in the migration file almost everything is memberships, dunno if it would be that quick to refactor all of it. But @DreyerChris, you do make sense. Your call

DreyerChris replied:
@RetroSabian Check what happens to the migrations if you change it. Yes it is really bad practice to give classes plural names. A `Membership` object is one instance of the class, therefore it should not have a plural name.

DreyerChris replied:
@RetroSabian Test it first in a separate branch to check if it breaks the migration. Then you can decide if you want to leave it or if you want to include it and manually fix the migrations

RetroSabian replied:
HI Chris, i redid the whole database, pease check it out

DreyerChris commented on Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Controllers/MembershipsController.cs:
```diff
@@ -0,0 +1,36 @@
+        }
+
+        [HttpPost("CreateEditMembership")]
+        public async Task Post([FromBody] Memberships Membership)
```
Standard is to make parameter names `camelCase`

RetroSabian replied:
will edit and use it this way from now on

DreyerChris commented on Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Controllers/MembershipsController.cs:
```diff
@@ -0,0 +1,36 @@
+            return await _db.MembershipsEntries.ToListAsync();
+        }
+
+        [HttpPost("CreateEditMembership")]
```
This function does not update memberships, it only creates them, why call it Edit as well?

RetroSabian replied:
This function will be edited modified completely, it is now to test front ends add functionality

DreyerChris replied:
Its never good to name functions based on planned future implementations, it will happen sometimes that you forget about it and then the name stays like that forever. It makes it especially hard for people trying to read your code, because the function acts like it does something when in reality it doesn't.

RetroSabian replied:
I do get this, however according to our wire frame this is one button, my task is to complete this whole integration, so will get to it as soon as i can confirm the create works

DreyerChris commented on Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs:
```diff
@@ -15,6 +15,11 @@ public Memberships()
         public string MembershipType { get; set; }
```
Usually its better to make "type" variables Enums, it makes it more robust because you only allow it to be a certain set of values

RetroSabian replied:
I was advised on this, have Enums on a to read list and will update accordingly

DreyerChris commented on Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs:
```diff
@@ -15,6 +15,11 @@ public Memberships()
         public string MembershipType { get; set; }
         public DateTime MembershipDuration { get; set; }
```
Why is this variable called `Duration` when it is a `DateTime`?  A `DateTime` indicates a point in time from which a duration can be calculated?

RetroSabian replied:
this is the intention, allowing the admin to create memberships and set a duration the membership runs for

grootendorste replied:
I think what @DreyerChris is saying is that a duration is usually an int or float or something. But with DateTime it's a specific set time (you'll have to compare it with a previous date first to get the duration). Otherwise maybe you should change the variable name, or at least add a comment to explain it.

DreyerChris replied:
Yeah exactly @grootendorste, I would suggest either making the duration a `int` for number of days or something or calling the field `MembershipExpiry` or something similar

RetroSabian replied:
okay int does make more sense

RetroSabian committed:
resolving PR comments

grootendorste reviewed (commented):
Looks good.

RetroSabian committed:
Redid Naming connection

-removed plural Entity Naming
-used Pascals naming convention
-Removed overly Descriptive naming

RetroSabian committed:
required change

RetroSabian committed:
Made Duration monetary

RetroSabian committed:
removed poor name

RetroSabian committed:
renamed Controllers to take the plural form

RetroSabian committed:
Updated README.md to manage migrations better

DreyerChris commented on Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Animation.cs:
```diff
@@ -0,0 +1,27 @@
+        public string FileLocation { get; set; }
+        public string Location { get; set; }
+        public string Description { get; set; }
+        public ICollection<UserAnimation> UserAnimation { get; set; }
```
You can give this property a plural name as it is a collection of UserAnimations. Do this for all the `ICollection` fields across all the classes

RetroSabian replied:
Updated

DreyerChris commented on Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Article.cs:
```diff
@@ -0,0 +1,28 @@
+        {
+            Initialise();
+        }
+
```
WS

RetroSabian replied:
Removed all WSs

DreyerChris commented on Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Organisation.cs:
```diff
@@ -0,0 +1,28 @@
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Description { get; set; }
+        public string URL { get; set;}
```
Usually its better practice to make URLs of type `uri`

RetroSabian replied:
Thanks didnt know about this

RetroSabian committed:
Fixed Naming Conventions

RetroSabian committed:
Changes to README.md
